src/middlewared/middlewared/plugins/usage.py

```diff
@@ -329,6 +329,37 @@ async def gather_pools(self, context):
     async def gather_services(self, context):
         return {'services': context['services']}
 
+    async def gather_nfs(self, context_unused):
+        num_clients = await self.middleware.call('nfs.client_count')
+        nfs_config = await self.middleware.call('nfs.config')
+        return {
+            'NFS': {
+                'enabled_protocols': nfs_config['protocols'],
+                'kerberos': nfs_config['v4_krb_enabled'],
+                'num_clients': num_clients,
+            }
+        }
+
+    async def gather_ftp(self, context_unused):
+        ftp_config = await self.middleware.call('ftp.config')
+
+        try:
+            res = subprocess.run(['netstat', '-4', '-6'], capture_output=True, text=True)
```

Comment:
A couple things:
1. `gather_ftp` is a coroutine but you're blocking main event loop with a synchronous fork+exec (subprocess).
2. I _really_ don't like this being called because busy systems could potentially have lots of entries and this is resolving ip addresses to dns names by default (which is also expensive).